n_helpers.c

```diff
@@ -348,25 +437,27 @@ const char * NoteBinaryTransmit(uint8_t * data, size_t dataLen, size_t bufLen, s
         return ERRSTR("buffer size exceeds available memory", c_mem);
     }
 
-    // Confirm buffer is large enough to in place encode as COBS
-    // NOTE: We need one additional byte beyond the encoded length to add a
-    //       newline character to the end of the buffer. This indicates the
-    //       end of transmission.
-    uint32_t encLen = cobsEncodedLength(data,dataLen);
-    if ((encLen + 1) > bufLen) {
-        NOTE_C_LOG_ERROR("buffer too small for encoding");
-        return ERRSTR("buffer too small for encoding", c_mem);
-    }
-
     // Calculate MD5
     char hashString[NOTE_MD5_HASH_STRING_SIZE] = {0};
     NoteMD5HashString(data, dataLen, hashString, NOTE_MD5_HASH_STRING_SIZE);
 
-    // Encode COBS data
+    // Shift the data to the end of the buffer. Next, we'll encode the data,
+    // outputting the encoded data to the front of the buffer.
     const size_t dataShift = (bufLen - dataLen);
     memmove(data + dataShift, data, dataLen);
-    encLen = cobsEncode(data + dataShift, dataLen, '\n', data);
-    data[encLen] = '\n';
+
+    // outLen holds the buffer size available for encoding. The -1 accounts for
+    // one byte of space we need to save for a newline to mark the end of the
+    // packet. When NoteBinaryEncode returns, outLen will hold the encoded
+    // length.
+    uint32_t outLen = bufLen - 1;
```

Comment:
> // outLen holds the buffer size available for encoding.

This comment is only necessary, because you have named the variable to match the parameter (context) of the function you are calling into, as opposed to the context of the current function in which it is being used.

Reply:
Oh yeah I meant to change this back to `encLen` per our conversation.